### ISSUE TITLE
Automatic tests stopped working with the release of PyMongo 4.0

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#3003]
```
tests/adapter/mongo/test_base.py::test_collection_stats ERROR            [  0%]

==================================== ERRORS ====================================
___________________ ERROR at setup of test_collection_stats ____________________

request = <SubRequest 'real_adapter' for <Function test_collection_stats>>
real_pymongo_client = MongoClient(host=['localhost:27017'], document_class=dict, tz_aware=False, connect=True)

    @pytest.fixture(scope="function")
    def real_adapter(request, real_pymongo_client):
        """Get an adapter connected to mongo database"""
        mongo_client = real_pymongo_client
    
        LOG.info("Connecting to database %s", REAL_DATABASE)
    
        database = mongo_client[REAL_DATABASE]
>       mongo_adapter = PymongoAdapter(database)

tests/conftest.py:682: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
scout/adapter/mongo/base.py:85: in __init__
    if database:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Database(MongoClient(host=['localhost:27017'], document_class=dict, tz_aware=False, connect=True), 'realtestdb')

    def __bool__(self):
>       raise NotImplementedError("Database objects do not implement truth "
                                  "value testing or bool(). Please compare "
                                  "with None instead: database is not None")
E       NotImplementedError: Database objects do not implement truth value testing or bool(). Please compare with None instead: database is not None
```

**Comment by northwestwitch** [comment]
To reproduce, create a fresh conda env with python 3.8, activate and 
1. Install scout (master branch) with `pip install -e .`
2. install libs required for running automatic tests with `pip install -r requirements-dev.txt `
3. Run pytest command `pytest -x`

**Comment by northwestwitch** [comment]
Ok they just released PyMongo 4.0. That could be the cause of the disruption: https://www.mongodb.com/community/forums/t/pymongo-4-0-released/134677

**Comment by northwestwitch** [comment]
Ok this can't be good:

![image](https://user-images.githubusercontent.com/28093618/144025017-f0ebbfe5-f8f8-44bb-9f4a-57cfbae6be15.png)
